src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -1056,10 +1057,12 @@ void WindowTransform::appendChunk(Chunk & chunk)
         // Another problem with Const columns is that the aggregate functions
         // can't work with them, so we have to materialize them like the
         // Aggregator does.
+        // Likewise, aggregate functions can't work with LowCardinality,
```

Comment:
Then why we didn't get exception in their factory methods?

Reply:
I was looking into this the other day (for a crazy idea of handling const columns in the aggregation functions) and saw it here: https://github.com/ClickHouse/ClickHouse/blob/be5bb7fe25f55be962350ed86c244dc40c9ceba9/src/Interpreters/Aggregator.cpp#L878

Both const and low cardinality columns are converted to full. Not very efficient but it works.

Reply:
I know. But the difference is that const does not affect the data type and LowCardinality does.
So, why AggregateFunctionFactory accepted LowCardinality?

Reply:
Aggregate function factory also converts all LC types to ordinary
https://github.com/ClickHouse/ClickHouse/blob/7a43a87f5b33c93db73d373ea83ae110b47cc1b8/src/AggregateFunctions/AggregateFunctionFactory.cpp#L70-L73

Reply:
Ok, let's fix the checks and merge.

Reply:
> Aggregate function factory also converts all LC types to ordinary

This is a tech debt after introducing LowCardinality.
Your code is doing one thing and expecting that some other unrelated code is doing the same thing.

It is very inconsistent.

It should work in the following way:
\- if Aggregate Function Factory has created an aggregate function with some argument types, we should be able to pass the values of these types inside the aggregate function.